neo4j-jdbc-bolt/src/main/java/org/neo4j/jdbc/utils/BoltNeo4jUtils.java

```diff
@@ -117,4 +120,30 @@ public static void closeSafely(AutoCloseable closeable, Logger logger) {
         }
     }
 
+    public static boolean hasResultSet(BoltNeo4jConnection connection, String cypher) {
+        try (final Session session = connection.newNeo4jSession();
+             final Transaction transaction = session.beginTransaction()) {
+            if (StringUtils.isBlank(cypher)) {
+                return false;
+            }
+            final String toUpperCase = cypher.trim().toUpperCase();
```

Comment:
this could break if you don't specify the locale.
For instance, a system locale set to Turkish would upper case `explain` to `EXPLAİN`, not `EXPLAIN`.

Reply:
done